core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -1221,6 +1221,11 @@ public final void addUser(String userName, Object credential, String[] roleList,
 		try {
 			roleList = UserCoreUtil
 					.combine(doGetInternalRoleListOfUser(userName, "*"), Arrays.asList(roleList));
+            		// If the newly created user has internal roles assigned from the UI wizard those internal roles
+            		// will be duplicated in the roles list. Duplcated roles are eliminated here.
+            		Set<String> rolesSet = new HashSet<String>(Arrays.asList(roleList));
```

Comment:
If the IDE is configured to use Java7, then this would have been indicated as a "warning" in the IDE. Please pay attention to those small things as much as possible.